allennlp/tools/quoref_eval.py

```diff
@@ -0,0 +1,108 @@
+import json
+from typing import Dict, Tuple, List, Any, Optional
+import argparse
+import numpy as np
+from allennlp.tools import drop_eval
+
+
+def _get_answers_from_data(annotations: Dict[str, Any]) -> Dict[str, List[str]]:
+    """
+    If the annotations file is in the same format as the original data files, this method can be used to extract a
+    dict of query ids and answers.
+    """
+    answers_dict: Dict[str, List[str]] = {}
+    for article_info in annotations["data"]:
+        for paragraph_info in article_info["paragraphs"]:
+            for qa_pair in paragraph_info["qas"]:
+                query_id = qa_pair["id"]
+                candidate_answers = [answer["text"] for answer in qa_pair["answers"]]
+                answers_dict[query_id] = candidate_answers
+    return answers_dict
+
+def evaluate_json(annotations: Dict[str, Any], predicted_answers: Dict[str, Any]) -> Tuple[float, float]:
```

Comment:
Why can't you use the drop script for this function (and the one above) also?

Reply:
Quoref's json format is the same as that of SQuAD, and DROP's is different. So `evaluate_json` needs to be different, and `evaluate_prediction_file` calls that function, so that needed to be rewritten as well. Mentioned this in the docstring.